src/binding.jl

```diff
@@ -386,6 +386,34 @@ function bind_pattern!(
         # disjunction: `(a | b)` where `a` and `b` are patterns.
         return bind_pattern!(location, Expr(:(||), source.args[2], source.args[3]), input, binder, assigned)
 
+    elseif is_expr(source, :tuple, 1) && is_expr(source.args[1], :parameters)
+        # named tuples (; x=p, y=q; z)
+        params = source.args[1]
+
+        conjuncts = BoundPattern[]
+
+        for param in params.args
+            field_name, pattern_source = parse_kw_param(param, source)
+
+            # Check that the field exists.
+            # We'd like to just test that input isa @NamedTuple($(field_names)...)
+            # But, NamedTuples are not covariant, so that test would fail if the tuple element is
+            # not inferred as exactly Any.
+            pattern = shred_where_clause(
```

Comment:
shredding is probably pointless. Its purpose is to decompose conjunctions and disjunctions and negations. But the target expression has none of those (by construction)